druid-shell/src/platform/x11/mod.rs

```diff
@@ -20,9 +20,22 @@
 
 pub mod application;
 pub mod clipboard;
-pub mod error;
 pub mod keycodes;
 pub mod menu;
 pub mod window;
 
+// TODO: This is just a placeholder, because we haven't removed all the other platform errors yet.
```

Comment:
I could go either way on this, but let me explain my rationale anyway:

The only point (AFAICT) of having an error enum is to allow users to match on it, and I have trouble seeing a use-case for matching on a platform-specific error from druid-shell.

One reason for this is that druid-shell actually handles most of its own errors, because it's in charge of the run loop. I had a quick look through the API, and I only saw `Application::new` and `WindowBuilder::build` as places that the user gets an error.

The platform-specific nature of a platform error also makes me think that it's unlikely to be handled by the user, together with the fact that -- looking at the current set of errors -- I don't see what they would do with one besides logging it and exiting.

Having said that, `anyhow::Error` *does* allow the user to get the underlying error if they really want it, because it supports downcasting. The main disadvantage compared to an error enum is ergonomics and discoverability.

Reply:
Not a lot of errors being surfaced outside of `druid-shell` is more about it just not having been done yet. However I agree with your point that it's unlikely anyone will want to match against platform specific errors.

However the platform code itself might want to match against its own errors. Error handling isn't always going to be exactly where it happens.

Also the Windows `Error::Hr(HRESULT)` continues to be very useful. There's no base error that can just be contexted. Thus I still don't see the platform specific `Error` type going away.

Reply:
Fair enough; I've reinstated the X11 error module.